oga/core.py

```diff
@@ -61,28 +61,31 @@ def from_file(cls, file_path: Optional[str] = None) -> Optional["Config"]:
             root_dir=section.get("root_dir", fallback=default.root_dir))
 
 
-async def search(session: aiohttp.ClientSession, base_query: str) -> AsyncGenerator[str, None]:
+async def search(session: aiohttp.ClientSession, base_query: str, page_limit: Optional[int] = None) -> AsyncGenerator[str, None]:
     page = 0
 
     async def fetch() -> bytes:
         url = f"{base_query}&page={page}"
         async with session.get(url) as response:
             return await response.read()
 
-    # Special case for first page since we may not continue
-    data = await fetch()
-    last_page = parse_last_search_page(data)
-    asset_ids = parse_search_results(data)
-    for asset_id in asset_ids:
-        yield asset_id
-
-    while page < last_page:
-        page += 1
+    if page_limit is None or page_limit > 0:
+        # Special case for first page since we may not continue
         data = await fetch()
+        last_page = parse_last_search_page(data)
+        if page_limit is not None and last_page > page_limit:
+            last_page = page_limit - 1
```

Comment:
Can this be simplified?  I'm not sure `last page > page_limit` and `page_limit - 1` produces the correct behavior when `last_page == page_limit`.

What about the following:
```
if page_limit is not None:
    last_page = min(last_page, page_limit - 1)
```

Reply:
You are right about it.
My mistake comes from originally misunderstanding the meaning of the variable.